osu.Framework/Platform/IWindow.cs

```
@@ -166,6 +166,16 @@ public interface IWindow : IDisposable
         /// </summary>
         Size ClientSize { get; }
 
+        /// <summary>
+        /// The minimum size of the window.
+        /// </summary>
+        Size MinSize { get; set; }
+
+        /// <summary>
+        /// The maximum size of the window.
+        /// </summary>
+        Size MaxSize { get; set; }
```

I'm not sure I wholly agree with putting this on `IWindow`, given that on a framework level it's describing a window on a platform-agnostic level, and these properties make little sense for mobile platforms. I'd just stick it on `SDL2DesktopWindow` and call it a day. [We already do casting to do desktop-specific stuff with the window game-side.](https://github.com/ppy/osu/blob/bb6e50eb27f8e8164e2c97b2a9b96bc62a45a4fd/osu.Desktop/OsuGameDesktop.cs#L139-L144)

This would also get rid of the misleading osuTK implementation that silently does nothing.

---

Yeah I was just not sure if casting to specific classes is the way here, I had it like you said initially.

---

The cleaner way to do this would be to adhere to interface segregation and have a `IDesktopWindow : IWindow` that extends the multiplatform capabilities with the specifics of desktop windows. But given that we have exactly one implementation right now that we're probably unlikely to switch out anytime soon, it's probably not worth the hassle.